libvast/src/system/node_control.cpp

```diff
@@ -45,9 +45,8 @@ spawn_at_node(caf::scoped_actor& self, const node_actor& node, invocation inv) {
         result = std::move(actor);
       },
       [&](caf::error& err) {
-        result
-          = caf::make_error(ec::unspecified, "failed to spawn '{}' at node: {}",
-                            inv.full_name, err);
+        result = vast::make_error(ec::unspecified, "failed to {} at node: {}",
```

Comment:
```suggestion
        result = make_error(ec::unspecified, "failed to {} at node: {}",
```

Same goes for everywhere else—not sure why we're explicitly specifying `vast::` from within the VAST code base now. Just feels very noisy to me.

(Obviously a style thing, so YMMV, and my opinion is not too strong on this. I just like consistency.)

Reply:
I'd like to drop the qualifier, but unfortunately it seems to be required. Thanks to ADL the compiler finds the overload from the `caf` namespace and prefers that because it doesn't require a conversion for the second argument.

If we change our own overload to
```c++
template <class... Args>
caf::error(ec code, Args&&... args) {
  if constexpr (sizeof...(Args) == 0)
    return caf::error{code};
  else
    return caf::error{code, fmt::format(std::forward<Args&&>(args)...)};
}
```
`fmt` complains about the format string isn't constant (which is wrong). I haven't looked if that can be worked around yet. Maybe you know?

Reply:
If you do the poisoning I suggested you get the correct one via ADL, always.

I think you just need a separate overload for the zero-argument version, and then have one that takes 1+ arguments, with the one being the {fmt} format string. I believe there is a `fmt::vformat*` family of functions you can use for inspiration on the format string argument.

Reply:
> If you do the poisoning I suggested you get the correct one via ADL, always.

No you don't, it will take the specialized one. The `= delete` has no effect on the overload set.

Reply:
`fmt::vformat` should work, but that isn't compile-time checked.

Reply:
vformat itself is not, but you want to check only once anyways, and that you do via the template arguments, by making sure you take a `fmt::format_string<Ts...>` and then matching `Ts...`.

Regarding ADL, shouldn't it always take the overload in namespace VAST because of `vast::ec` anyways? I don't understand why it wouldn't work without the full qualification.

Reply:
> Regarding ADL, shouldn't it always take the overload in namespace VAST because of vast::ec anyways?

ADL only adds to the overload set, for `vast::ec` this doesn't make a difference because we're in namespace VAST anyways. But it doesn't have an effect on overload resolution, if `caf::make_error` is visible and a better match it will be selected.

Reply:
It shouldn't be a better match here, though, if the first argument to the function is in the vast namespace.

Reply:
Best idea I have been able to come up with is to name our version of the function differently. (`gen_error`?)

Reply:
We could also consider just putting our overload into the namespace `caf`.

Reply:
That won't change the fact that our overload needs a conversion so the other one will still be preferred.